cmd/opm/alpha/veneer/composite.go

```diff
@@ -0,0 +1,128 @@
+package veneer
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/operator-framework/operator-registry/alpha/veneer/composite"
+)
+
+func newCompositeVeneerRenderCmd() *cobra.Command {
+	var (
+		veneer        composite.Veneer
+		output        string
+		containerTool string
+		validate      bool
+		configFile    string
+	)
+	cmd := &cobra.Command{
+		Use: "composite composite-veneer-file",
+		Short: `Generate a file-based catalog from a single 'composite veneer' file
+When FILE is '-' or not provided, the veneer is read from standard input`,
+		Long: `Generate a file-based catalog from a single 'composite veneer' file
+When FILE is '-' or not provided, the veneer is read from standard input`,
+		Args: cobra.MaximumNArgs(1),
```

Comment:
From the doc:
>opm alpha render-veneer composite -f catalog/config.yaml -c .../catalogs.yaml

We need to have two inputs to this process:
1. the composite veneer itself
2. the catalogs config to define valid builders, source opm environment, and working dir for generated artifacts

So wouldn't we have two inputs and `cobra.MinimumNArgs(2)`?

Reply:
I don't think so. The way I have it setup is to run like so:
```
opm alpha render-veneer composite catalogs.yaml -c catalog/config.yaml
```
The catalogs.yaml file is fed in via a positional argument and config.yaml file can be specified by the `--composite-config, -c` flag with it's default value being to look for `catalog/config.yaml`

Reply:
Right, and the alternative format of this can use the input redirect as 
`opm alpha render-veneer composite - -c catalog/config.yaml <catalogs.yaml`
which gets a little confusing.  
Isn't it clearer to drop the pipe capability since we have two mandatory inputs and no way of knowing which one the user will want to pipe in?

I got thrown initially by the use of the positional vs. enumerated arg passing.

Reply:
I guess that makes sense but I would suspect that someone wanting to do a form of input redirection may end up wanting to organize their command a bit more like:
```
opm alpha render-veneer composite -c catalog/config.yaml - <catalogs.yaml
```
(i.e organize it such that all the flags are declared first and then do the file or input redirect at the end)

I'm also not against just forcing the use of flags and files and setting default values for the flags if that is what we think is the best UX flow. I was just trying to keep this as similar to the other `opm alpha render-veneer ...` subcommands as possible

> Isn't it clearer to drop the pipe capability since we have two mandatory inputs and no way of knowing which one the user will want to pipe in?
This is also a very good point with the perspective that the current implementation would only let you pipe the catalogs.yaml file and not the catalog/config.yaml equivalent.

Reply:
Went ahead and updated the implementation for this in commit https://github.com/operator-framework/operator-registry/pull/1054/commits/606acd5eef65c62ae77a29e0eddaaf55b8332a86

Reply:
However I forgot to change the args :facepalm: